### ISSUE TITLE
ReachingDefinition analysis fails because of the stack pointer being `Undefined`

### ISSUE THREAD

**Issue by Pamplemousse** [angr/angr#1537]
Trying to run the "reaching definition" analysis from the angr repo (via the test: [tests/test_reachingdefinitions.py](https://github.com/angr/angr/blob/master/tests/test_reachingdefinitions.py)), I toggled the comments [line 25-26](https://github.com/angr/angr/blob/master/tests/test_reachingdefinitions.py#L25-L26), and ran `nosetests tests/test_reachingdefinitions.py`.

It is failing, and here is the end of the stacktrace I obtain:

<details>
  <summary>Toggle error details</summary>

```
ERROR: test_reachingdefinitions.test_reaching_definition_analysis(<Project /home/pamplemousse/Workspace/angr/angr/tests/../../binaries/tests/x86_64/al
l>, <Function main (4196038)>, {})                                                                                                                   
----------------------------------------------------------------------                                                                               
Traceback (most recent call last):                                                                                                                   
  File "/home/pamplemousse/Workspace/angr/venv/lib/python3.5/site-packages/nose/case.py", line 198, in runTest                                       
    self.test(*self.arg)                                                                                                                             
  File "/home/pamplemousse/Workspace/angr/angr/tests/test_reachingdefinitions.py", line 21, in run_reaching_definition_analysis                      
    _ = project.analyses.ReachingDefinitions(func, kb=tmp_kb)                                                                                        
  File "/home/pamplemousse/Workspace/angr/angr/angr/analyses/analysis.py", line 109, in __call__                                                     
    oself.__init__(*args, **kwargs)                                                                                                                  
  File "/home/pamplemousse/Workspace/angr/angr/angr/analyses/reaching_definitions/reaching_definitions.py", line 370, in __init__                    
    self._analyze()                                                                                                                                  
  File "/home/pamplemousse/Workspace/angr/angr/angr/analyses/forward_analysis.py", line 588, in _analyze                                             
    self._analysis_core_graph()                                                                                                                       
  File "/home/pamplemousse/Workspace/angr/angr/angr/analyses/forward_analysis.py", line 611, in _analysis_core_graph                                 
    changed, output_state = self._run_on_node(n, job_state)                                                                                          
  File "/home/pamplemousse/Workspace/angr/angr/angr/analyses/reaching_definitions/reaching_definitions.py", line 464, in _run_on_node                
    state = engine.process(state, block=block, fail_fast=self._fail_fast)                                                                            
  File "/home/pamplemousse/Workspace/angr/angr/angr/analyses/reaching_definitions/engine_vex.py", line 30, in process                                
    self._process(state, None, block=kwargs.pop('block', None))                                                                                      
  File "/home/pamplemousse/Workspace/angr/angr/angr/engines/light/engine.py", line 58, in _process                                                   
    self._process_Stmt()                                                                                                                             
  File "/home/pamplemousse/Workspace/angr/angr/angr/engines/light/engine.py", line 78, in _process_Stmt                                              
    getattr(self, handler)()                                                                                                                         
  File "/home/pamplemousse/Workspace/angr/angr/angr/analyses/reaching_definitions/engine_vex.py", line 486, in _handle_function                      
    raise TypeError('Invalid type %s for stack pointer.' % type(sp_addr).__name__)                                                                   
TypeError: Invalid type Undefined for stack pointer. 
```
</details>

It seems that the `SimEngineRDVEX`'s state does not have the right `register_definitions` content, as [angr/analyses/reaching_definitions/engine_vex.py#L471](https://github.com/angr/angr/blob/master/angr/analyses/reaching_definitions/engine_vex.py#L471) contains `{<Definition {Atom:<Reg 48<8>>, Codeloc:<0x4006f8 id=0x4006c6[81]>, Data:DataSet<64>: {<Undefined>}}>}`.

I have tried out couple of other binaries from the [angr/binaries repo](https://github.com/angr/binaries) without much more success, and always trigger the same bug.

---

<details>
  <summary>Here is my angr environment report</summary>

```
angr environment report
=============================
Date: 2019-04-26 13:22:10.763897
Running in virtual environment at /home/pamplemousse/Workspace/angr/venv
Platform: linux-x86_64
Python version: 3.5.6 (default, Aug  2 2018, 13:05:50) 
[GCC 7.3.0]
######## angr #########
Python found it in /home/pamplemousse/Workspace/angr/angr/angr
Pip version angr 8.19.4.5
Git info:
	Current commit 7e78a25b4a82278679748188ea5255f13ddb7b1e from branch master
Could not resolve tracking branch or remote info!
######## ailment #########
Python found it in /home/pamplemousse/Workspace/angr/venv/lib/python3.5/site-packages/ailment
Pip version ailment 8.19.4.5
Couldn't find git info
######## cle #########
Python found it in /home/pamplemousse/Workspace/angr/cle/cle
Pip version cle 8.19.4.5
Git info:
	Current commit 2cdd1523bc8d3b6a9d0a1368150129bd2bfa9ccc from branch master
	Checked out from remote origin: https://github.com/angr/cle.git
######## pyvex #########
Python found it in /home/pamplemousse/Workspace/angr/pyvex/pyvex
Pip version pyvex 8.19.4.5
Git info:
	Current commit c418edc1146982b2a0579bf56e5993c1c7046b19 from branch master
	Checked out from remote origin: https://github.com/angr/pyvex.git
######## claripy #########
Python found it in /home/pamplemousse/Workspace/angr/claripy/claripy
Pip version claripy 8.19.4.5
Git info:
	Current commit 4ed61924880af1ea8fb778047d896ec0156412a6 from branch master
	Checked out from remote origin: https://github.com/angr/claripy.git
######## archinfo #########
Python found it in /home/pamplemousse/Workspace/angr/archinfo/archinfo
Pip version archinfo 8.19.4.5
Git info:
	Current commit fca8d2bf23044caf207e160b25a3809cbadfbec6 from branch master
	Checked out from remote origin: https://github.com/angr/archinfo.git
######## z3 #########
Python found it in /home/pamplemousse/Workspace/angr/venv/lib/python3.5/site-packages/z3
Pip version not found!
Couldn't find git info
######## unicorn #########
Python found it in /home/pamplemousse/Workspace/angr/venv/lib/python3.5/site-packages/unicorn
Pip version unicorn 1.0.1
Couldn't find git info
######### Native Module Info ##########
z3: <CDLL '/home/pamplemousse/Workspace/angr/venv/lib/python3.5/site-packages/z3/lib/libz3.so', handle 22843a0 at 0x7fbf555ab4e0>
unicorn: <CDLL '/home/pamplemousse/Workspace/angr/venv/lib/python3.5/site-packages/unicorn/lib/libunicorn.so', handle fb3510 at 0x7fbf53fdad68>
angr: <CDLL '/home/pamplemousse/Workspace/angr/angr/angr/lib/angr_native.so', handle 1b8bd30 at 0x7fbf57a94ef0>
pyvex: <cffi.api._make_ffi_library.<locals>.FFILibrary object at 0x7fbf55971a90>
```
</details>

**Comment by Pamplemousse** [comment]
Ah, I think I finally located the problem!
Well, it's not really a problem, but more of a poor usage I made of the `ReachingDefinitions`...

It seems that setting the parameter `init_func=True` when calling `project.analyses.ReachingDefinitions()` ends up working fine.

From what I have debugged, not setting it to true does not initialize the abstract values.
Hence, when applying the VEX IR transformations to that unitialized state, all operations on stack pointer's values result into setting an `Undefined` value, hence the encountered error...

Will give some thinking to find a way that would avoid future developers to think that this is a bug.
